news_events/etl/sloan_webinars.py

```diff
@@ -0,0 +1,217 @@
+                        },
+                    },
+                ]
+            },
+            "aura.context": {
+                "mode": "PROD",
+                "fwuid": "FWUID_VAR",
+                "app": "siteforce:communityApp",
+                "loaded": {
+                    "APPLICATION@markup://siteforce:communityApp": "APPID_VAR",
+                    "COMPONENT@markup://instrumentation:o11ySecondaryLoader": "",
+                },
+                "dn": [],
+                "globals": {},
+                "uad": False,
+            },
+            "aura.pageURI": "/s/webinars",
+            "aura.token": None,
+        }
+    )
+    .replace("None", "null")
+    .replace("False", "false")
+    .replace("+", "")
+    .replace("%27", "%22")
+)
+
+
+def extract() -> dict:
+    """
+    Extract JSON from Sloan's Webinar listing
+
+    Returns:
+        dict: JSON data from Sloan blog post request
+    """
+    session = requests.Session()
+    content = str(session.get(SLOAN_WEBINAR_URL).content)
+    fwuid = re.findall(r"fwuid%22%3A%22([^%]+)%22", content)[0]
+    appId = re.findall(r"siteforce%3AcommunityApp%22%3A%22([^%]+)%", content)[0]
+    session.headers["Content-Type"] = "application/x-www-form-urlencoded"
+    return session.post(
+        SLOAN_WEBINAR_POST_URL,
+        data=SLOAN_WEBINAR_POST_DATA.replace("FWUID_VAR", fwuid).replace(
+            "APPID_VAR", appId
+        ),
+    ).json()
+
+
+def extract_event_image(image_data: dict) -> tuple[dict, dict]:
+    """
+    Extract the image data from the Open Learning Event.
+
+    Args:
+        image_data (dict): The image data
+
+    Returns:
+        tuple[dict, dict]: The image text metadata and image source metadata
+
+    """
+    altText = image_data.get("altText", "")
+    imagePath = image_data.get("url", "")
+    if imagePath:
+        return {
+            "url": urljoin(SLOAN_WEBINAR_BASE_URL, imagePath),
+            "alt": altText if altText else "",
+            "description": altText if altText else "",
+        }
+    else:
+        return None
+
+
+def transform_item(event_data: dict) -> dict:
+    """
+    Transform item from Sloan School of Management blog
+
+    Args:
+        event_data (dict): raw JSON data for a single webinar post
+
+    Returns:
+        dict: Transformed data for a single webinar post
+
+    """
+    attributes = event_data.get("contentNodes", {})
+    guid = event_data["contentKey"]
+    dt = event_data.get("publishedDate")
+    text_date = attributes.get("Image_Text", {}).get("value", "")
+    try:
+        dt_utc = (
+            timefhuman(text_date)[0]
+            .replace(tzinfo=ZoneInfo("US/Eastern"))
+            .astimezone(UTC)
+            if dt
+            else None
+        )
+    except:  # noqa: E722
+        dt_utc = (
+            parser.parse(dt).replace(tzinfo=ZoneInfo("US/Eastern")).astimezone(UTC)
+            if dt
+            else None
+        )
```

Comment:
So in summary, if the event date cannot be parsed, I think it might be better to not publish it (and log an error or warning) than to use the publish date instead.

Reply:
I got feedback on this from peter as well. I think ignoring unparsable items is the way to go.